src/sc/proofverifier.cpp

```diff
@@ -65,6 +74,48 @@ namespace libzendoomc{
         return true;
     }
 
+    ScFieldElement CalculateCertDataHash(const CScCertificate& cert) {
+        // TODO Replace with hash of merkel tree supplied by SC.
+        ScFieldElement certDataHash;
+        certDataHash.SetHex(cert.GetHash().GetHex());
+        return certDataHash;
+    }
+
+    bool CalculateCumulativeCertDataHash(const ScFieldElement& prevCumulativeHash, const ScFieldElement& prevCertHash, ScFieldElement& CumulativeHashOutput) {
```

Comment:
Could we return the ScFieldElement CumulativeHashOutput instead of bool and test outside if it is valid or not?
I personally hate this assembly-like style; but I can live with it